From d10da28968d2c8b8add4dd2cf77c7aa928979222 Mon Sep 17 00:00:00 2001
From: Sebastien Guiriec <s-guiriec@ti.com>
Date: Tue, 8 Mar 2011 18:06:40 +0100
Subject: OMAP4: hwmod: Configure McPDM in SMART IDLE WAKE UP mode

As McPDM can be an OCP initiator the IP should be configured in
SMART IDLE WAKE UP capable. It will allow to have McPDM direct
path (ABE bypass) working when PM is enabled.

Change-Id: Ia762994ec530af5606f2240e656cc2ba44cabc7b
Signed-off-by: Sebastien Guiriec <s-guiriec@ti.com>
---
 arch/arm/mach-omap2/omap_hwmod_44xx_data.c |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/arch/arm/mach-omap2/omap_hwmod_44xx_data.c b/arch/arm/mach-omap2/omap_hwmod_44xx_data.c
index 92a3b66..8e63ec4 100644
--- a/arch/arm/mach-omap2/omap_hwmod_44xx_data.c
+++ b/arch/arm/mach-omap2/omap_hwmod_44xx_data.c
@@ -3244,7 +3244,8 @@ static struct omap_hwmod_class_sysconfig omap44xx_mcpdm_sysc = {
 	.sysc_offs	= 0x0010,
 	.sysc_flags	= (SYSC_HAS_EMUFREE | SYSC_HAS_RESET_STATUS |
 			   SYSC_HAS_SIDLEMODE | SYSC_HAS_SOFTRESET),
-	.idlemodes	= (SIDLE_FORCE | SIDLE_NO | SIDLE_SMART),
+	.idlemodes	= (SIDLE_FORCE | SIDLE_NO | SIDLE_SMART |
+			   SIDLE_SMART_WKUP),
 	.sysc_fields	= &omap_hwmod_sysc_type2,
 };
 
-- 
1.6.0.4

